src/read_write/node_writer.rs

```diff
@@ -164,3 +196,137 @@ impl CubeNodeWriter {
         self.layer_writers[0].bytes_written() as i64 / 3
     }
 }
+
+pub struct RawNodeWriter {
+    xyz_writer: DataWriter,
+    layer_writers: Vec<DataWriter>,
+    stem: PathBuf,
+}
+
+impl NodeWriter for RawNodeWriter {
+    fn write(&mut self, p: &Point) {
+        p.position.write_le(&mut self.xyz_writer);
+        p.color.write_le(&mut self.layer_writers[0]);
+        if let Some(i) = p.intensity {
+            ensure_intensity_layer(&mut self.layer_writers, &self.stem);
+            i.write_le(&mut self.layer_writers[1]);
+        };
+    }
+}
+
+impl RawNodeWriter {
+    pub fn new(path: impl Into<PathBuf>) -> Self {
+        let stem: PathBuf = path.into();
+        let xyz_writer =
+            DataWriter::new(&stem.with_extension(NodeLayer::Position.extension())).unwrap();
+        let layer_writers =
+            vec![DataWriter::new(&stem.with_extension(NodeLayer::Color.extension())).unwrap()];
+        Self {
+            xyz_writer,
+            layer_writers,
+            stem,
+        }
+    }
+}
+
+pub struct PlyNodeWriter {
+    writer: DataWriter,
+    point_count: usize,
+}
+
+impl NodeWriter for PlyNodeWriter {
+    fn write(&mut self, p: &Point) {
+        if self.point_count == 0 {
+            let mut layers = vec![NodeLayer::Position, NodeLayer::Color];
+            if p.intensity.is_some() {
+                layers.push(NodeLayer::Intensity);
+            }
+            self.create_header(&layers);
+        }
+
+        p.position.write_le(&mut self.writer);
+        p.color.write_le(&mut self.writer);
+        if let Some(i) = p.intensity {
+            i.write_le(&mut self.writer);
+        };
+
+        self.point_count += 1;
+    }
+}
+
+impl Drop for PlyNodeWriter {
+    fn drop(&mut self) {
+        self.writer.write_all(b"\n").unwrap();
+        if self.writer.seek(SeekFrom::Start(4 + 32 + 15)).is_ok() {
```

Comment:
Do you think it would be cleaner to compute a Seek offset in create_header and store it in the PlyNodeWriter?

Reply:
I updated it to use two consts, one for the part until the number of vertices, one for the number of vertices themselves.